### DISCUSSION TITLE
Difference between gitoxide and pijul?

### DISCUSSION THREAD

**Discussion by ghost** [GitoxideLabs/gitoxide#402] [Q&A] [answered]
Hello, I have a question, what is the difference between gitoxide and pijul(https://nest.pijul.com/pijul/pijul)?

**Comment by Byron** [comment, marked as answer]
For all I know, `pijul` is an entirely new and somewhat unique revision control system written in Rust whereas `gitoxide` is an implementation of the `git` revision control system in Rust.

Maybe a good entry-point to learn more about Pijul is [their FAQ](https://pijul.org/faq).

**Comment by burdges** [comment]
`git` stores snapshots of the underlying data, but supports compression across snapshots.  Although wonderful, `git` often needs human help resolving conflicts, or scary tools [`git rerere`](https://www.git-scm.com/book/en/v2/Git-Tools-Rerere), so automated `git` usages winds up rare. 

`pijul` is a new dvcs that works mostly with patches instead of snapshots, but solves some performance problems of `darcs`, the other patch based dvcs.  It appears pijul merge always succeed, which might leave source in an unusable state, but then resolving the conflict once creates a patch that could resolve the conflict for reverse merge operation, aka "rebase and merge are the same operation in pijul".

A prioir, I think resolving the conflicts in-tree like this makes `pijul` sound nicer for automates use cases, like some distributed hackmd or whatever, but not sure about the specifics.  `pijul` is not really battle tested and little tooling exists though.

Aside from the [FAQ](https://pijul.org/faq) the pijul manual has informative [theory](https://pijul.com/manual/theory.html) and [channels](https://pijul.com/manual/workflows/channels.html) sections, and good blog posts from [Nov 2020 blog post](https://pijul.org/posts/2020-11-07-towards-1.0/) and [Dec 2020](https://pijul.org/posts/2020-12-19-partials/).

**Reply by ghost** [reply]
Nice explanation, thank you very much
